modules/nf-core/clipkit/main.nf

```diff
@@ -0,0 +1,45 @@
+process CLIPKIT {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/clipkit:2.3.0--pyhdfd78af_0':
+        'biocontainers/clipkit:2.3.0--pyhdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(aln)
+
+    output:
+    tuple val(meta), path("*.clipkit"), emit: clipkit
+    path "versions.yml"               , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    clipkit \\
+        $args \\
+        $aln
+
```

Comment:
Maybe rename the clipkit file here to `${prefix}.clipkit` that way you are 100% sure it has the correct name

Reply:
The default functionality is for the tool to just add ".clipkit" to the input file name. We can either overwrite that within the script by passing the -o argument as suggested ( e.g. ${prefix}.clipkit ) or leave it to the user to do that from a nextflow.config through -args. I left it like this because I thought that just adding a suffix to the filename would be the most wanted functionality. Else to achieve this, the user would need to pass the filename as a meta.id, right?

Which is the preferred nf-core way to do this?

Reply:
I'd prefer the `-o` option to stay consistent with all other modules. The input file name can sometimes be changed by another process in a way that the user doesn't want it. This way you can have full control over the file name in the nf-core best-practice way :)